public/app/features/alerting/unified/components/rule-editor/labels/LabelsField.tsx

```diff
@@ -0,0 +1,477 @@
+import { css, cx } from '@emotion/css';
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { Controller, FormProvider, useFieldArray, useForm, useFormContext } from 'react-hook-form';
+
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { Button, Field, InlineLabel, Input, LoadingPlaceholder, Space, Stack, Text, useStyles2 } from '@grafana/ui';
+import { useDispatch } from 'app/types';
+
+import { labelsApi } from '../../../api/labelsApi';
+import { usePluginBridge } from '../../../hooks/usePluginBridge';
+import { useUnifiedAlertingSelector } from '../../../hooks/useUnifiedAlertingSelector';
+import { fetchRulerRulesIfNotFetchedYet } from '../../../state/actions';
+import { SupportedPlugin } from '../../../types/pluginBridges';
+import { RuleFormValues } from '../../../types/rule-form';
+import AlertLabelDropdown from '../../AlertLabelDropdown';
+import { AlertLabels } from '../../AlertLabels';
+import { NeedHelpInfo } from '../NeedHelpInfo';
+
+import { AddButton, RemoveButton } from './LabelsButtons';
+
+const useGetOpsLabelsKeys = (skip: boolean) => {
+  const { currentData, isLoading: isloadingLabels } = labelsApi.endpoints.getLabels.useQuery(undefined, {
+    skip,
+  });
+  return { loading: isloadingLabels, labelsOpsKeys: currentData };
+};
+const useGetCustomLabels = (dataSourceName: string): { loading: boolean; labelsByKey: Record<string, Set<string>> } => {
```

Comment:
Should we name it to `useGetAlertRulesLabels`? `customLabels` seems to be a bit confusing, isn't it?

Reply:
agreed! it was the naming already used in this part, but you're right, it's confusing 👍